### ISSUE TITLE
Building vmtx from features.fea

### ISSUE THREAD

**Issue by NightFurySL2001** [googlefonts/fontmake#1013]
Currently when including vmtx in features.fea it raises an error that the functionality is unsupported:
```
INFO:ufo2ft:Building OpenType tables
INFO:ufo2ft.outlineCompiler:The copyright was normalized for storage in the CFF table and consequently some characters were dropped: 'Copyright 2022-2023 The Unbounded Sans Project Authors https:github.comNightFurySL2001unbounded-sans
Copyright 2020 The Dela Sans Project Authors https:github.comsyakuzenDelaGothic'
fontmake: Error: In 'UnboundedSans.ufo': Compiling UFO failed: UnboundedSans.ufo\features.fea:28:1: "table vmtx" is not supported
```

Will there be support for [vmtx in features.fea](https://adobe-type-tools.github.io/afdko/OpenTypeFeatureFileSpecification.html#9.h) soon? Also vhea support might crash with #586 so that might need to be looked into too.

**Comment by punchcutter** [comment]
I opened a PR last year for this https://github.com/fonttools/fonttools/pull/2704, but I knew at the time it is undesirable to do it this way. I only matched what makeotf does internally. For UFO sources you can avoid writing the feature and have it derived from the info in the UFO. The PR discussion explains how to get vmtx without putting it in the features file.

**Comment by NightFurySL2001** [comment]
That is true, but for editors that doesn't support vmtx natively it is better and easier to override in features.fea since overriding features.fea is quite normal after generating UFO from source (I think?), while also making it easier by not needing to specify the metrics for all the glyphs. Otherwise it is another script to maintain before building the font.

This actually remind me of the other thing that setting the keys like in #586 but no height and public.verticalOrigin in individual glyphs makes the glyphs have 0 height in vmtx. I'm not sure if that's intended or a bug.

**Comment by ChiuMing-Neko** [comment]
@NightFurySL2001 @anthrotype

I agree that it would be beneficial for `fontmake` to support reading `vmtx` feature syntax and applying it during the build process, which is crucial for CJK font building process, rather than throwing an error from `ufo2ft` when everything else is going smoothly.

For those who use `fontmake` in a build pipeline and need to apply `vmtx` definitions from feature files, here is a suggested workaround:

Use `fonttools.ufoLib` to open the UFO instance and write the following information directly to each glyph:

-   Set `glyph.height` for the `VertAdvanceY` value. This corresponds to `<advance height="{value}"/>` in the `.glif` file.
-   Set `glyph.lib["public.verticalOrigin"]` for the `VertOriginY` value. This corresponds to the `public.verticalOrigin` key in the glyph's `lib`.

Then, as noted in issue #586, ensure that the UFO's `fontinfo.plist` contains the following attributes. These are required for `fontmake` to generate the `vmtx` table:

-   `openTypeVheaVertTypoAscender`
-   `openTypeVheaVertTypoDescender`
-   `openTypeVheaVertTypoLineGap`
-   `openTypeVheaCaretSlopeRise`
-   `openTypeVheaCaretSlopeRun`
-   `openTypeVheaCaretOffset`

Without these attributes, the variable OTF generated by `fontmake` may fail to be recognized by some software (such as the Windows Font Viewer) due to a missing `vmtx` table when a `vhea` table is present. This can happen even if the font passes fontbakery validation.
